components/scripts/gradle/03-validate-local-build-caching-different-locations.sh

```diff
@@ -158,8 +173,34 @@ execute_second_build() {
 }
 
 fetch_build_cache_metrics() {
-  read_build_scan_metadata
-  fetch_and_read_build_scan_data build_cache_metrics_only "${build_scan_urls[@]}"
+  if [ "$build_scan_publishing_mode" == "on" ]; then
+    read_build_scan_metadata
+    fetch_and_read_build_scan_data build_cache_metrics_only "${build_scan_urls[@]}"
+  else
+    find_build_scan_dumps
+    read_build_scan_dumps
+  fi
+}
+
+find_build_scan_dumps() {
+  find_build_scan_dump "first"
+  find_build_scan_dump "second"
+}
+
+find_build_scan_dump() {
+  local build_name="$1"
+  local build_dir="${EXP_DIR}/$build_name-build_${project_name}"
+  build_scan_dump="$(find "$build_dir" -maxdepth 1 -type f -regex '^.*build-scan-.*-.*-.*-.*\.scan' | sort | tail -n 1)"
+  if [ -z "$build_scan_dump" ]; then
+    die "ERROR: No Build Scan dump found for the $build_name build"
+  fi
+  eval "${build_name}_build_scan_dump=\${build_scan_dump}"
```

Comment:
-1: If we wanted to avoid the eval here, we could create a `build_scan_dumps` array instead. This is what we do for normal Build Scans.

You could declare the array like this:

```
build_scan_dumps=()
```

And then you could add to the array here like this:

```
build_scan_dumps+="${build_scan_dump}"
```

To refer to one of the build scan dumps when invoking the Java utility:

```
"${build_scan_dumps[0]" "${build_scan_dumps[1]}"
```

Reply:
Great suggestion. Addressed by 5f81c814a56131e1ab6f418c27f4fe7660aaf8b3.